code/datums/traits.dm

```diff
@@ -171,6 +173,10 @@
 				T = trait_instance
 			if(T.afterlife_blacklisted && inafterlifebar(owner))
 				return
+			// TODO: probably fix syntax
+			var/resolved_role = owner?.mind?.assigned_role || src.mind_role_fallback
+			if (T.preventAddTrait(owner, resolved_role))
```

Comment:
Wondering if we should have a way to skip this check? e.g. admins trying to add traits that aren't typically available.

Reply:
Eh, future requirement. If an admin wants to apply the only trait this gets affected by for now, they can just go activate one of their bioeffects and reinforce it using admin tools.

Reply:
I solved it ANYWAY